integration/tests/cook/test_cli.py

```diff
@@ -892,10 +892,43 @@ def test_show_progress_message(self):
             instance = util.wait_until(lambda: cli.show_json(uuids, self.cook_url)[1][0]['instances'][0],
                                        lambda i: 'progress' in i and 'progress_message' in i)
             self.assertEqual(99, instance['progress'])
-            self.assertEqual("We're so close!", instance['progress_message'])
+            self.assertEqual("We are so close!", instance['progress_message'])
             cp = cli.show(uuids, self.cook_url)
             self.assertEqual(0, cp.returncode, cp.stderr)
-            self.assertIn("99% We're so close!", cli.stdout(cp))
+            self.assertIn("99% We are so close!", cli.stdout(cp))
+            cp = cli.tail(uuids[0], 'stdout', self.cook_url, '--lines 100')
+            self.logger.debug(f'Contents of stdout: {cli.decode(cp.stdout)}')
+            self.assertIn("99 We are so close!", cli.decode(cp.stdout))
+            self.assertIn('Command exited with status 0', cli.decode(cp.stdout))
+            self.assertIn('Executor completed execution', cli.decode(cp.stdout))
+
+    def test_show_progress_message_custom_progress_file(self):
+        executor = util.get_job_executor_type(self.cook_url)
+        cp, uuids = cli.submit('\'touch progress.txt && '
+                               'echo "Hello World" >> progress.txt && '
+                               'echo "progress: 99 We are so close!" >> progress.txt && '
+                               'echo "Done" >> progress.txt\' ',
+                               self.cook_url,
+                               submit_flags=f'--executor {executor} '
+                                            '--env EXECUTOR_PROGRESS_OUTPUT_FILE_ENV=PROGRESS_OUTPUT_FILE '
+                                            '--env PROGRESS_OUTPUT_FILE=progress.txt')
+        self.assertEqual(0, cp.returncode, cp.stderr)
+        util.wait_for_job(self.cook_url, uuids[0], 'completed')
+        self.assertEqual(0, cp.returncode, cp.stderr)
+        cp, jobs = cli.show_json(uuids, self.cook_url)
+        self.assertEqual(0, cp.returncode, cp.stderr)
+        self.assertEqual(executor, jobs[0]['instances'][0]['executor'])
+        if executor == 'cook':
+            instance = util.wait_until(lambda: cli.show_json(uuids, self.cook_url)[1][0]['instances'][0],
+                                       lambda i: 'progress' in i and 'progress_message' in i)
```

Comment:
Would it make sense to create a helper function for this since it's duplicated above?

Reply:
Created `__wait_for_progress_message`